### PR TITLE
API-1455: Add the debug log for EventsApi SendRequestError

### PR DESCRIPTION
**Pull request by pyaillet**
**Description (for Contributor and Core Developer)**

Add the debug log for EventsApi SendRequestError

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/back/Application/Webhook/Service/EventsApiDebugLogger.php [hunk 1]
```diff
@@ -66,6 +68,41 @@ public function logLimitOfEventsApiRequestsReached(): void
...
+
+    public function flushLogs(): void
+    {
+        if (0 === count($this->buffer)) {
+            return;
+        }
+
+        $this->repository->bulkInsert($this->buffer);
+        $this->buffer = [];
+    }
```
Comment (tseho) [inline src/Akeneo/Connectivity/Connection/back/Application/Webhook/Service/EventsApiDebugLogger.php:104]: Are you sure ?

Comment (pyaillet) [reply src/Akeneo/Connectivity/Connection/back/Application/Webhook/Service/EventsApiDebugLogger.php]: I'm sure it's a leftover from a rebase error 👍 
Thanks for spotting it

**Review by pchasle — COMMENTED** [review]
You are missing a unit test (phpspec) on `GuzzleWebhookClient.php` to test that the error is logged.

File: src/Akeneo/Connectivity/Connection/back/Application/Webhook/Service/EventsApiDebugResponseErrorLogger.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+interface EventsApiDebugResponseErrorLogger
+{
+    /**
+     * @param string $connectionCode
+     * @param array<WebhookEvent> $events
+     * @param string $url
+     * @param int $statusCode
+     * @param array<array<string>> $headers
+     */
+    public function logResponseError(string $connectionCode, array $events, string $url, int $statusCode, array $headers): void;
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/back/Application/Webhook/Service/EventsApiDebugResponseErrorLogger.php:22]:
```suggestion
    public function logEventsApiRequestFailed(string $connectionCode, array $events, string $url, int $statusCode, array $headers): void;
```

and `logEventsApiRequestSucceed` for the futur one ?

Comment (tomglvng) [reply src/Akeneo/Connectivity/Connection/back/Application/Webhook/Service/EventsApiDebugResponseErrorLogger.php]: done in https://github.com/akeneo/pim-community-dev/pull/14016

File: src/Akeneo/Connectivity/Connection/back/tests/Unit/spec/Infrastructure/Webhook/Client/GuzzleWebhookClientSpec.php [hunk 3]
```diff
@@ -116,6 +126,49 @@ public function it_sends_webhook_requests_in_bulk(SendApiEventRequestLogger $sen
         Assert::assertEquals($signature, $request->getHeader(RequestHeaders::HEADER_REQUEST_SIGNATURE)[0]);
     }
 
+    public function it_should_log_an_error_when_an_error_occurs(SendApiEventRequestLogger $sendApiEventRequestLogger, EventsApiDebugResponseErrorLogger $responseErrorLogger): void
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/back/tests/Unit/spec/Infrastructure/Webhook/Client/GuzzleWebhookClientSpec.php:129]:
```suggestion
    public function it_logs_a_failed_events_api_request(SendApiEventRequestLogger $sendApiEventRequestLogger, EventsApiDebugResponseErrorLogger $responseErrorLogger): void
```

**Review by pchasle — APPROVED** [review]

File: src/Akeneo/Connectivity/Connection/back/Application/Webhook/Service/EventsApiDebugLogger.php [hunk 4]
```diff
@@ -66,6 +67,31 @@ public function logLimitOfEventsApiRequestsReached(): void
...
+    public function logEventsApiRequestFailed(string $connectionCode, array $webhookEvents, string $url, int $statusCode, array $headers): void
+    {
+        $this->repository->persist([
+            'timestamp' => $this->clock->now()->getTimestamp(),
+            'level' => EventsApiDebugLogLevels::ERROR,
+            'message' => 'The endpoint returned an error.',
+            'connection_code' => $connectionCode,
+            'event_subscription_url' => $url,
+            'status_code' => $statusCode,
+            'headers' => $headers,
```
Comment (tseho) [inline src/Akeneo/Connectivity/Connection/back/Application/Webhook/Service/EventsApiDebugLogger.php:86]: should go inside the context.

**Review by tseho — APPROVED** [review]
GTM after fix

**Pull request merged**